lib/logstash/filters/json.rb

```diff
@@ -83,7 +83,7 @@ def filter(event)
     end
 
     if @target
-      event.set(@target, parsed)
+      event.set(event.sprintf(@target), parsed)
```

Comment:
Good point @jordansissel! I think it'd be nice to support pulling a field from the parsed object, but how would that addressing even work since we also need to support indirect addressing based on existing fields?

I'm leaning towards keeping the existing behavior but documenting it (I forgot about that in this first version of the patch). Selecting the target field based on fields from the parsed object is possible via an extra mutate operation afterwards so we won't be blocking anything with this limitation. Sounds good to you?